heartbeat/monitors/logger/logger.go

```diff
@@ -18,34 +18,43 @@
 package logger
 
 import (
-	"errors"
+	"encoding/json"
+	"fmt"
+	"sync"
 	"time"
 
 	"github.com/elastic/beats/v7/libbeat/beat"
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
+var mtx = sync.Mutex{}
 var eventLogger *logp.Logger = nil
 
 const ActionMonitorRun = "monitor.run"
 
-type durationLoggable struct {
+const META_STEP_COUNT = "__HEARTBEAT_STEP_COUNT__"
+
+type DurationLoggable struct {
```

Comment:
This probably doesn't need to be exported

Reply:
For stuff like this I'm not too worried, the codebase isn't really re-used elsewhere